src/main/java/io/deepstream/utils/AckTimeoutRegistry.java

```diff
@@ -0,0 +1,167 @@
+package io.deepstream.utils;
+
+import io.deepstream.ConnectionChangeListener;
+import io.deepstream.IDeepstreamClient;
+import io.deepstream.constants.Actions;
+import io.deepstream.constants.ConnectionState;
+import io.deepstream.constants.Event;
+import io.deepstream.constants.Topic;
+import io.deepstream.message.Message;
+
+import java.util.*;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.TimerTask;
+
+public class AckTimeoutRegistry implements ConnectionChangeListener {
+
+    private Map<String, AckTimeoutTask> register;
+    private Timer timer;
+    private Topic topic;
+    private long timeoutDuration;
+    private IDeepstreamClient client;
+    private ConnectionState state;
+    private LinkedBlockingQueue<AckTimeoutTask> ackTimers;
+
+    /**
+     * The registry for all ack timeouts.
+     *
+     * @param client The client it sends errors to
+     * @param topic The topic for all acks in the register
+     * @param timeoutDuration The timeout duration for acks in milliseconds
+     */
+    public AckTimeoutRegistry( IDeepstreamClient client, Topic topic, long timeoutDuration ) {
+        this.client = client;
+        this.register = new HashMap<String, AckTimeoutTask>();
+        this.ackTimers = new LinkedBlockingQueue<AckTimeoutTask>();
+        this.timer = new Timer();
+        this.topic = topic;
+        this.timeoutDuration = timeoutDuration;
+
+        this.state = client.getConnectionState();
+        this.client.addConnectionChangeListener( this );
+    }
+
+    /**
+     * Clears the ack timeout for a message.
+     *
+     * @param message The message received to remove the ack timer for
+     */
+    public void clear( Message message ) {
+        String uniqueName = this.getUniqueName( Actions.getAction( message.data[ 0 ] ), message.data[ 1 ] );

```

Comment:
Getting action just to revert it back to string might be a bit odd, just noticed